pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -104,26 +104,8 @@ func (f *ring2Factory) PrintSuccess(shortOutput bool, out io.Writer, resource, n
 	}
 }
 
-func (f *ring2Factory) PrintObject(cmd *cobra.Command, isLocal bool, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error {
-	// try to get a typed object
-	_, typer := f.objectMappingFactory.Object()
-	gvks, _, err := typer.ObjectKinds(obj)
-
-	if err != nil {
-		return err
-	}
-	// Prefer the existing external version if specified
-	var preferredVersion []string
-	if gvks[0].Version != "" && gvks[0].Version != runtime.APIVersionInternal {
-		preferredVersion = []string{gvks[0].Version}
-	}
-
-	mapping, err := mapper.RESTMapping(gvks[0].GroupKind(), preferredVersion...)
-	if err != nil {
-		return err
-	}
-
-	printer, err := f.PrinterForMapping(ExtractCmdPrintOptions(cmd, false), mapping)
+func (f *ring2Factory) PrintObject(cmd *cobra.Command, isLocal bool, obj runtime.Object, out io.Writer) error {
```

Comment:
I want a really nasty check in this function to start (we may move it).  If we're printing something here, it's either an object in our scheme or an unstructured.  I want to use the CLI scheme as a typer (NOT the factory) and if the obj is recognized and has `""` or `runtime.APIVersionInternal` as its version, I want to error on "sent an internal object %v(gvk) to print". That will let us chase down problems with callers.

We **must** avoid printing anything but name, wide, or "" using internal objects.  We can try the check here first.

Reply:
Added a check here https://github.com/kubernetes/kubernetes/pull/59907/commits/703dabe149fd1cf755e397f5890e6b67a589fb83